### ISSUE TITLE
Performance of (re-)generating large SGraphs

### ISSUE THREAD

**Issue by mlange05** [ecmwf-ifs/loki#254]
With the new Scheduler (PR #213) approaching completion, one of the remaining issues flagged was the performance of building and re-building large SGraph objects, eg. for EC-physics. As a result we reverted to caching the SGraph and only rebuilding it when needed (after discovery and parsing). This issue is to track the necessary performance improvements needed to re-enable fast on-the-fly re-builds.

As a note, a quick initial investigation showed that some of the parser-class caching in `make_complete` is not quite working as intended yet, but a "quick fix" had severe knock-on effects. Moreover, even with this, there is probably still a significant amount of time spend using `FindNodes` to (re-)define the dependencies for each SGraph item, causing re-builds to do many traversals per rebuild, which could be improved.

**Comment by reuterbal** [comment]
@mlange do you consider this sufficiently resolved with #684?

**Comment by mlange05** [comment]
Yes, the remaining time is not negligible, but given we're re-building the full graph from scratch there is little to be done without major surgery. It does look much better now though, so yes, I think this can be closed.
